rest/taskana-rest-spring/src/main/java/pro/taskana/user/rest/UserController.java

```diff
@@ -0,0 +1,126 @@
+@EnableHypermediaSupport(type = HypermediaType.HAL)
+public class UserController {
+  private final UserService userService;
+  private final UserRepresentationModelAssembler assembler;
+
+  @Autowired
+  UserController(UserService userService, UserRepresentationModelAssembler assembler) {
+    this.userService = userService;
+    this.assembler = assembler;
+  }
+
+  /**
+   * This endpoint retrieves a User.
+   *
+   * @title Get a User
+   * @param userId the id of the requested User
+   * @return the requested User
+   * @throws UserNotFoundException if the id has not been found
+   */
+  @GetMapping(URL_USERS_ID)
+  @Transactional(readOnly = true, rollbackFor = Exception.class)
+  public ResponseEntity<UserRepresentationModel> getUser(@PathVariable String userId)
+      throws UserNotFoundException {
+    User user = userService.getUser(userId);
+
+    return ResponseEntity.ok(assembler.toModel(user));
+  }
+
+  /**
+   * This endpoint creates a User.
+   *
+   * @title Create a User
+   * @param repModel the User which should be created
+   * @return the inserted User
+   * @throws InvalidArgumentException if the id has not been set
+   * @throws UserAlreadyExistException if a User with id } is already existing
+   * @throws NotAuthorizedException if the current user is no admin or business-admin
+   */
+  @PostMapping(RestEndpoints.URL_USERS)
+  @Transactional(rollbackFor = Exception.class)
+  public ResponseEntity<UserRepresentationModel> createUser(
+      @RequestBody UserRepresentationModel repModel)
+      throws InvalidArgumentException, UserAlreadyExistException, NotAuthorizedException {
+    User user = assembler.toEntityModel(repModel);
+    user = userService.createUser(user);
+
+    return ResponseEntity.status(HttpStatus.CREATED).body(assembler.toModel(user));
+  }
+
+  /**
+   * This endpoint updates a User.
+   *
+   * @title Update a User
+   * @param userId the id of the User to update
+   * @param repModel the User with the updated fields
+   * @return the inserted User
+   * @throws InvalidArgumentException if the id has not been set
+   * @throws UserNotFoundException if a User with id is not existing in the database
+   * @throws NotAuthorizedException if the current user is no admin or business-admin
+   */
+  @PutMapping(URL_USERS_ID)
+  @Transactional(rollbackFor = Exception.class)
+  public ResponseEntity<UserRepresentationModel> updateUser(
+      @PathVariable(value = "userId") String userId, @RequestBody UserRepresentationModel repModel)
+      throws InvalidArgumentException, UserNotFoundException, NotAuthorizedException {
+    if (!userId.equals(repModel.getUserId())) {
+      throw new InvalidArgumentException(
+          String.format(
+              "UserId '%s' of the URI is not identical"
+                  + " with the userId '%s' of the object in the payload.",
+              userId, repModel.getUserId()));
+    }
+    User user = assembler.toEntityModel(repModel);
+    userService.updateUser(user);
```

Comment:
Please save the returned user into a variable here and put it into the return statement below. Currently the passed user is returned

Reply:
Currently, the update method does not change any of the fields. It just checks whether a User with same id is already existing in the database. Thus, I made the updateUser(...) not return an object because it would be the same as passed to the method.

Reply:
That's true but imagine if we add a timestamp or something like that in the future, then this will be fatal. I'd rather just be consistent with the rest of our code. Also the call is made anyways, so no performance gain here.

Reply:
Oh I overread, that the method doesnt even return anything right now...hm okay

Reply:
You're right 👍